Feat/ksp mock-bank 거래내역 통계(feign) 연동 및 통계 조회 추가

KSP0321 opened the pull request:
## 📌 관련 이슈
- mock-bank 거래내역 통계(feign) 연동
-AccountQueryService/Controller 통계 조회 추가
-user entity 수정으로 오류난 testcode 수정

## ✨ 변경 사항


## 📷 Postman 
-실행 사진은 mock에서 확인하시면 될 것 같습니다!

## ✅ 체크리스트
- [ ] 관련 이슈에 연결됨
- [ ] 기능이 정상적으로 동작함
- [ ] 코드 리뷰를 받았음
- [ ] 불필요한 로그/코드 제거됨

## 💬 기타 참고 사항
- 테스트 방법, 추가 설명 등 필요한 경우 작성해 주세요.

KSP0321 committed:
feat : account 관련 sns/sqs 구독 수정

KSP0321 committed:
feat : account 관련 sns/sqs 구독 수정2

KSP0321 committed:
feat : 회원가입시 salary 필드 추가

KSP0321 committed:
feat : .salary(request.getSalary()) 추가

KSP0321 committed:
feat : 거래내역 총수익, 총지출, 특정 기간 내 평균 수익, 지출 정보 호출

KSP0321 committed:
test : user 에 salary 필드 추가로 관련 테스트 코드 오류 수정

KSP0321 committed:
feat : userupdate 부분에서 salary validation 추가

KSP0321 committed:
merge dev

KSP0321 committed:
refactor : merge dev 중 추가된 엔티티로 생긴 충돌 수정

KSP0321 committed:
feat : feat : 각자 account-created-queue-dev- 뒤에 자기 이니셜 붙이면 되겠습니다.

KSP0321 committed:
feat : 각자 account-created-topic-dev- 뒤에 자기 이니셜 붙이면 되겠습니다.

KSP0321 committed:
feat : 임시 userId 인증(헤더 비교 등) 코드 제거

byeongtaek12 commented on src/main/java/org/example/lifechart/domain/account/service/AccountQueryService.java:
```diff
@@ -27,18 +30,21 @@ public AccountResponse getAccount(Long userId) {
+    public TransactionStatResponse getUserTransactionStats(Long userId, YearMonth startYM, YearMonth endYM) {
+        TransactionStatRequest req = new TransactionStatRequest(startYM, endYM);
+        MockBankApiResponse<TransactionStatResponse> response = accountClient.getTransactionStats(userId, req);
 
```
혹시 컨트롤러에서 req를 받아서 그대로 서비스에 넘겨주시지 않고 req에서 꺼내서 startYM과 endYM으로 빼서 넘겨준다음 다시 새로 같은 객체를 만드시는 이유가 무엇인지 궁금합니다.

KSP0321 replied:
저도 제가 왜 저렇게 짰는지 모르겠습니다;;;
수정했습니다!

byeongtaek12 commented on src/main/java/org/example/lifechart/domain/user/dto/AccountCreatedEvent.java:
```diff
@@ -4,13 +4,15 @@
     private String email;
-    private String nickname;
     private String userName;
+    private BigDecimal salary;
```
금융이나 통화 같은 정확한 계산이 필요할 때 double이나 float 대신 사용한다는 것을 배워갑니다. 처음 보는 타입이네요

byeongtaek12 commented on src/main/resources/secrets/application-secret.example.properties:
```diff
@@ -11,14 +11,19 @@
+# account (dev)
+aws.url.sqs.account=https://sqs.ap-northeast-2.amazonaws.com/885378802418/account-created-queue-dev-(ksp/lhw/jkb/obt/sch/lhj)
+aws.arn.sns.account=arn:aws:sns:ap-northeast-2:885378802418:account-created-topic-dev-(ksp/lhw/jkb/obt/sch/lhj)
+
```
👍

byeongtaek12 reviewed (approved):
코멘트 남긴 거 답변만 해주시면 될 것 같습니다. 수고하셨습니다.

rlqor9688 commented on src/main/java/org/example/lifechart/domain/account/controller/AccountQueryController.java:
```diff
@@ -28,27 +28,30 @@ public class AccountQueryController {
     private final AccountQueryService accountQueryService;
 
     @GetMapping("/{userId}")
-    public ResponseEntity<ApiResponse<AccountResponse>> getAccount(@PathVariable Long userId, @AuthenticationPrincipal CustomUserPrincipal principal) {
-
-        // 권한 체크 (본인만 가능)
-        if (!principal.getUserId().equals(userId)) {
-            throw new CustomException(ErrorCode.NO_AUTHORIZATION);
-        }
```
로그인 유저 검증은 다른 곳에서 처리해서 로직을 수정하신 걸까요?

KSP0321 replied:
아 pr 기다리다가 작업한다고 지우고 pr 덮어씌워졌습니다

rlqor9688 reviewed (commented):
고생하셨습니다 승표님! 코멘트 남겼습니다!

KSP0321 committed:
feat: 계좌·거래내역 조회 시 본인 권한 검증 로직
- AuthUtil.validateUserAccess(userId) 메서드로 권한 검증 일원화
- 컨트롤러(혹은 서비스)에서 Feign 호출 전에 반드시 아래 코드 삽입

rlqor9688 reviewed (approved):
확인했습니다! 고생하셨습니다 승표님~ 병택님 코멘트만 답변해주시고 머지하시면 될 것 같습니다!

KSP0321 committed:
refactor : TransactionStat 조회 컨트롤러 및 서비스 파라미터 전달 방식 개선
